test/workbox-strategies/sw/test-Strategy.mjs

```diff
@@ -0,0 +1,331 @@
+/*
+  Copyright 2020 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+import {timeout} from 'workbox-core/_private/timeout.mjs';
+import {Strategy} from 'workbox-strategies/Strategy.mjs';
+import {spyOnEvent} from '../../../infra/testing/helpers/extendable-event-utils.mjs';
+
+
+class FetchStrategy extends Strategy {
+  _handle(request, handler) {
+    return handler.fetch(request);
+  }
+}
+
+class LongWaitUntilStrategy extends Strategy {
+  _handle(request, handler) {
+    handler.waitUntil(timeout(200));
+    return new Response('generated response');
+  }
+}
+
+class EmptyStrategy extends Strategy {
+  _handle(request, handler) {
+    return new Response('generated response');
+  }
+}
+
+class ErrorStrategy extends Strategy {
+  _handle(request, handler) {
+    handler.waitUntil(Promise.reject(new Error('waitUntil error')));
+    return new Response('generated response');
+  }
+}
+
+class ExtendingStrategy extends FetchStrategy {
+  constructor(options) {
+    super(options);
+    this._newProperty = options.newProperty;
+  }
+  async _handle(request, handler) {
+    handler.waitUntil(timeout(100));
+    return await handler.cacheMatch(request) ||
+        await super._handle(request, handler);
+  }
+}
+
+function generateOptions() {
+  const request = new Request('/strategy-request');
+  const event = new FetchEvent('fetch', {request});
+  spyOnEvent(event);
+
+  return {request, event};
+}
+
+describe(`Strategy`, function() {
+  const sandbox = sinon.createSandbox();
+
+  beforeEach(async function() {
+    const keys = await caches.keys();
+    await Promise.all(keys.map((key) => caches.delete(key)));
+    sandbox.restore();
+  });
+
+  afterEach(async function() {
+    const keys = await caches.keys();
+    await Promise.all(keys.map((key) => caches.delete(key)));
+    sandbox.restore();
+  });
+
+  xdescribe('constructor()', function() {
```

Comment:
I guess this is disabled via the `x` prefix. Was  that intentional?

Reply:
Unintentional, reverted.